### PR TITLE
Godwin Assignments

### PR DESCRIPTION
**Pull request by Headieh**

### CODE REVIEW

File: students/headieh_godwin/lesson03/list_lab.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+print(fruit[int(response1)-1])
+#Add another fruit to the beginning of the list using "+" and display the list.
+fruit = ['Mango'] + fruit
+print(fruit)
+#Add another fruit to the beginning of the list using insert() and display the list.
+fruit.insert(0, 'kiwi')
+print(fruit)
+#Display all the fruits that begin with "P", using a for loop.
+for x in fruit:
+    if x.startswith('P'):
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/list_lab.py]: User of upper() would be good here

File: students/headieh_godwin/lesson03/list_lab.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        print(x)
+
+#SERIES 2
+
+#Using the list created in series 1 above:
+#Display the list.
+print(fruit)
+#Remove the last fruit from the list.
+fruit2=fruit[:-1]
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/list_lab.py]: Missing whitespace around assignment operator

File: students/headieh_godwin/lesson03/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+#Display the list.
+print(fruit2)
+#Ask the user for a fruit to delete, find it and delete it.
+response2 = input("which fruit would you like to delete?")
+fruit2 = [x for x in fruit2 if x != response2]
+print(fruit2)
+#(Bonus: Multiply the list times two. Keep asking until a match is found.
+#Once found, delete all occurrences.)
+fruit2=fruit2*2
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/list_lab.py]: Missing whitespace here too

File: students/headieh_godwin/lesson03/list_lab.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+response3 = input("which fruit would you like to delete?")
+if response3 not in fruit2:
+    run = True
+else:
+    run= False
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/list_lab.py]: Missing whitespace after run

File: students/headieh_godwin/lesson03/list_lab.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+print(fruit)
+#for each fruit in the list (making the fruit all lowercase).
+fruit3 = [item.lower() for item in fruit]
+
+for i in fruit3:
+#Ask the user for input displaying a line like "Do you like apples?"
+    response4 = input("Do you like {}".format(i))
+#For any answer that is not "yes" or "no":
+    if response4 not in ['no', 'yes']:
+        run=True
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/list_lab.py]: Whitespace here

File: students/headieh_godwin/lesson03/list_lab.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+#prompt the user to answer with one of those two values 
+            print("please write 'no' or 'yes'")
+            response4 = input("Do you like {}".format(i))
+            if response4 not in ['no', 'yes']:
+                run=True
+            else:
+                run=False
+#For each "no", delete that fruit from the list.
+    if response4 == 'no':
+        fruit3 =[x for x in fruit3 if x != i]
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/list_lab.py]: Nice!

File: students/headieh_godwin/lesson03/slicing_lab.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+def case5(x):
+    y = list(x)
+    amt = int(round(len(y)/3))
+    first = x[:-amt]#all but last 3rd (first 3rd+middle third)
+    last = x[-amt:]#(approximately or exactly last 3rd)
+    return last+first
+
+assert case5(a_string) == 'tringthis is a s'
+assert case5(a_tuple) == (5, 32, 2, 54, 13, 12)
+assert case5(b_tuple) == (6, 3, 1, 99, 2, 54, 13, 12, 5, 32, 9, 8, 7)
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/slicing_lab.py]: Looks good - nice assert statements!

File: students/headieh_godwin/lesson03/string_formatting_lab.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+#TASK 4
+#take tuple and reorder and reformat
+##start with: ( 4, 30, 2017, 2, 27)
+##end with '02 27 2017 04 30'
+def neworder(in_tuple):
+    temp =  ', '.join(['{:02d}']*5).format(*in_tuple) 
+    temp = temp.split(", ")
+    last = temp[-2:]
+    first = temp[:2]
+    temp[-2:]=first
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/string_formatting_lab.py]: Make sure and keep whitespacing consistent.

**Comment by Headieh** [comment]
Didn't see the linter comment till it was too late - I'll look into that before the next lesson

**Comment by natasha-aleksandrova** [comment]
@charlesrobison looks like mailroom has been added - can you review that as well please?

**Comment by natasha-aleksandrova** [comment]
@Headieh yes you want to make sure you sync with upstream and pull in the changes in master. Let us know if you need help with that.

File: students/headieh_godwin/lesson03/mailroom.py [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+        elif resp == 'Create a Report':
+            report()
+        else:
+            print('Something went wrong')
+            break
+
+def original_prompt():
+    """ Prompt for a user selection """
+    questions = [
+        inquirer.List('action',
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/mailroom.py]: Okay, inquirer is cool!

File: students/headieh_godwin/lesson03/mailroom.py [hunk 10]
```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+import inquirer
+
+donors = [['Karen', (20,20,100)],
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/mailroom.py]: Missing whitespace after commas in the tuples

File: students/headieh_godwin/lesson03/mailroom.py [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+    """ Prompt for a user selection """
+    questions = [
+        inquirer.List('action',
+                      message="What would you like to do? ",
+                      choices=['Send a Thank You', 'Create a Report', 'quit'])]
+    answers = inquirer.prompt(questions)
+    return(answers['action'])
+
+def email(x,y):
+    print( f"""
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/mailroom.py]: No whitespace before f string

File: students/headieh_godwin/lesson03/mailroom.py [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+def original_prompt():
+    """ Prompt for a user selection """
+    questions = [
+        inquirer.List('action',
+                      message="What would you like to do? ",
+                      choices=['Send a Thank You', 'Create a Report', 'quit'])]
+    answers = inquirer.prompt(questions)
+    return(answers['action'])
+
+def email(x,y):
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/mailroom.py]: Missing whitespace after comma here

File: students/headieh_godwin/lesson03/mailroom.py [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+        response2 = input("Please enter a donation amount ")
+        #response2 = int(response2)
+        if (response2 == "") or (float(response2) < 0.01):
+            print("Please enter a value greater than 0")
+        elif float(response2) > 100000:
+            print('Please enter a smaller value')
+        else:
+            new_donor = [response1, float(response2)]
+            donors.append(new_donor)
+            email(new_donor[0],new_donor[1])
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/mailroom.py]: Whitespace after comma here too

File: students/headieh_godwin/lesson03/mailroom.py [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+        #response2 = int(response2)
+        if (response2 == "") or (float(response2) < 0.01):
+            print("Please enter a value greater than 0")
+        elif float(response2) > 100000:
+            print('Please enter a smaller value')
+        else:
+            filters = [response1 in names for names in donors]
+            filters = [i for i, x in enumerate(filters) if x][0]
+            wo_d = donors[filters][1]
+            response2= float(response2)
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/mailroom.py]: Keep whitespace consistent

File: students/headieh_godwin/lesson03/mailroom.py [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+            else:
+                w_d = (wo_d,)+(response2,)
+            donors[filters][1]=w_d
+            email(response1, response2)
+
+#Report
+def report():
+    """ prints a report of donors """
+    #print(donors) FOR TESTING
+    col_labels=["Donor Name","Total Given","Num Gifts", "Average Gift"]
```
Comment (charlesrobison) [inline students/headieh_godwin/lesson03/mailroom.py]: Whitespace issues here too

**Pull request merged**